test/martha/martha_v3.test.js

```diff
@@ -107,99 +135,272 @@ test.serial('martha_v3 resolves a valid DOS-style url', async (t) => {
+    getJsonFromApiStub.onCall(1).resolves(bondAccessTokenResponse);
+    getJsonFromApiStub.onCall(2).resolves(drsAccessUrlResponse);
+    const response = mockResponse();
+    await marthaV3(
+        mockRequest(
+            {
+                body: {
+                    url: 'drs://dg.712C/fa640b0e-9779-452f-99a6-16d833d15bd0',
+                    fields: ['accessUrl'],
+                }
+            },
+        ),
+        response,
+    );
+    t.is(response.statusCode, 200);
+    const result = response.send.lastCall.args[0];
+    sinon.assert.callCount(getJsonFromApiStub, 3); // Bond was not called to retrieve the googleServiceAccount
+    t.deepEqual(
+        { ...result },
+        mask(bdcDrsMarthaResult(googleSAKeyObject, drsAccessUrlResponse), 'accessUrl'),
+    );
+
+    const requestedBondAccessTokenUrl = getJsonFromApiStub.getCall(1).args[0];
+    const accessTokenUrlMatches = requestedBondAccessTokenUrl.match(bondAccessTokenUrlRegEx);
+    t.truthy(accessTokenUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
+    const expectedAccessTokenProvider = 'fence';
+    const actualAccessTokenProvider = accessTokenUrlMatches[2];
+    t.is(actualAccessTokenProvider, expectedAccessTokenProvider);
+
+    t.is(
+        getJsonFromApiStub.getCall(2).args[0],
+        `https://${config.HOST_BIODATA_CATALYST_STAGING}/ga4gh/drs/v1/objects` +
+        '/dg.712C/fa640b0e-9779-452f-99a6-16d833d15bd0/access/gs',
+    );
+    t.is(getJsonFromApiStub.getCall(2).args[1], `Bearer ${bondAccessTokenResponse.token}`);
+});
+
+test.serial('martha_v3 calls the correct endpoints when only the fileName is requested and the metadata contains a name', async (t) => {
+    const drsResponse = bdcDrsResponseCustom({
+        name: 'HG01131.final.cram.crai',
+        access_url: { url: bdcDrsResponse.access_methods[0].access_url.url },
+        access_id: bdcDrsResponse.access_methods[0].access_id,
+    });
+    const drsAccessUrlResponse = mockGcsAccessUrl(bdcDrsResponse.access_methods[0].access_url.url);
+    getJsonFromApiStub.onCall(0).resolves(drsResponse);
+    const response = mockResponse();
+    await marthaV3(
+        mockRequest(
+            {
+                body: {
+                    url: 'drs://dg.712C/fa640b0e-9779-452f-99a6-16d833d15bd0',
+                    fields: ['fileName'],
+                }
+            },
+        ),
+        response,
+    );
+    t.is(response.statusCode, 200);
+    const result = response.send.lastCall.args[0];
+    sinon.assert.callCount(getJsonFromApiStub, 1); // File name available from the metadata
+    t.deepEqual(
+        { ...result },
+        mask(bdcDrsMarthaResult(googleSAKeyObject, drsAccessUrlResponse), 'fileName'),
+    );
+});
+
+test.serial('martha_v3 calls the correct endpoints when only the fileName is requested and the metadata contains a gs url', async (t) => {
+    const drsResponse = bdcDrsResponseCustom({
```

Comment:
This results in an unmodified `bdcDrsResponse`. As a result, I think this behavior is already covered by `martha_v3 parses BDC response correctly`.

Reply:
Yes, true today, and happy to remove this as a separate test for now. If/when BDC changes what fields get populated we can change both `bdcDrsResponse` and `bdcDrsResponseCustom` at that time.